extensions/binding/org.eclipse.smarthome.binding.mqttbroker/src/main/java/org/eclipse/smarthome/binding/mqttbroker/internal/NetworkUtils.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.mqttbroker.internal;
+
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.net.util.SubnetUtils;
+
+/**
+ * Network utility functions for determining all interfaces and assigned IP addresses.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class NetworkUtils {
```

Comment:
I assume this has already been discussed:
Why not moving this class to our net utils?

Reply:
Because the net core library is a holy ground and I didn't want to touch it :smile:.  And because I don't want to write tests for it, nor adapting all the extensions and core parts that have similar methods, I guess :D

Reply:
Sorry, these arguments don't count. Please check what can be used from `NetUtil` resp. `NetworkAddressService` and see what additional methods make sense there in general.

Reply:
This is understandable and saddening at the same time 😅

Reply:
This means, going through every ESH extension and OH binding and looking for network utils and all the implications like debating about the correct API, writing all the documentation and then changing all those addons. I'm not paid by the Eclipse Foundation, so I happily refuse this, but I can create a bug report instead :)

Reply:
Isn't DRY a best practice for any coder?

Reply:
Definitely. But there is always a balance of how time can be used. The guy that implemented the Network Service for providing the user configured network interface name, could have done this for example, because it was much more related.

Reply:
Not sure if we are talking about the same thing. What we are asking is whether any of the methods here might be useful in general and thus would be better of in the core framework instead of in the binding. Looking at the class name and the methods, it appears that there is nothing binding-specific about it and the class and method names are closely resembling the ones that already exist.

And the "guy that implemented the Network Service" did so merely to provide a feature that can be used across multiple bindings (and he even [adapted them all](https://github.com/openhab/openhab2-addons/pull/2559/files) to make use of that feature; we are not asking here for any such refactoring on any other bindings).

Reply:
Added one method to net/core: https://github.com/eclipse/smarthome/pull/4172